esx_service/utils/auth_api.py

```diff
@@ -190,7 +195,8 @@ def modify_privileges(privileges, allow_create, volume_maxsize_in_MB, volume_tot
     logging.debug("modify_privileges: allow_create=%s, volume_maxsize_in_MB=%s, volume_totalsize_in_MB=%s",
                   allow_create, volume_maxsize_in_MB, volume_totalsize_in_MB)
     if allow_create is not None:
-        if allow_create == "True":
+        # allow_create has been valided. It is either "true" or "false"
+        if allow_create == "true":
```

Comment:
Well, it's even worse then. Please make sure it's type is consistent and not fluctuates between invocations.

I'd suggest to keep as boolean always - i..e one entry call `validate_bool_string` or `conert_to_bool` (built-in bool() won't work for you)  to make sure it's True or False (boolean)  and use str() when needed (though Python will do it for you on print

if you want it to be string always, fine as well. (use `type(var) == bool` to check incoming stuff from vmodl.)